### DISCUSSION TITLE
A question about FastKernelSurvivalSVM()

### DISCUSSION THREAD

**Discussion by Lamgayin** [sebp/scikit-survival#476] [Q&A]
Hello Author,

I would like to inquire about the predict() function in FastKernelSurvivalSVM, which states that "Lower ranks indicate shorter survival, higher ranks indicate longer survival." 

However, why is it necessary to multiply val by -1 when rank_ratio equals 1? 

Does this mean that the smaller the output value, the longer the predicted survival time; and conversely, the larger the output value, the shorter the predicted survival time? In this case, does the predict function return a ranking, where lower values correspond to longer survival times and higher values correspond to shorter survival times? Is this understanding correct?

thx a lot

**Comment by sebp** [comment]
Actually, the description of predict is incorrect. If the rank_ratio is 
1, it returns risk scores. That means samples with a higher risk score 
are predicted to experience the event before samples with a lower risk 
score.

Multiplication by -1 achieves this, because the objective function 
assigns higher values to samples with higher survival time.
